ci-operator/step-registry/baremetal/lab/ipi/install/baremetal-lab-ipi-install-commands.sh

```diff
@@ -153,6 +153,22 @@ for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
   prepare_bmc "${bmc_address}" "${bmc_user}" "${bmc_pass}"
 done
 
+# if loadbalancer type is UserManaged, it's mean using external LB, 
+# then keepalived and haproxy will not deployed, but coredns still keep 
+if [ ${LB_TYPE} = "UserManaged" ]; then
+  # Delete the exist api/ingress vips
+  yq e --inplace 'del(.platform.baremetal.apiVIP,.platform.baremetal.ingressVIP)' ${SHARED_DIR}/install-config.yaml
+
+  #Get the external VIP and updated to install-config.yaml
+  APIVIP=$(yq .api_vip ${SHARED_DIR}/external_vips.yaml)
```

Comment:
Thanks, I updated the script as your suggestions

> What are the IPs we should refer to in the HAProxy container? Currently, we use the internal VIPs... With this feature, should we list all the internal master nodes' IPs?

yes, I think so if the apiVIP can route to the master node's IPs and ingress workers.

Reply:
ok, you also need to change the configuration of the loadbalancer then.

This is kind of similar to the upi one now. 

The current IPI one uses the Internal VIPs to configure the LB: https://github.com/openshift/release/blob/519acedbb12e8f27cd576157e6a72860a35f5f0c/ci-operator/step-registry/baremetal/lab/ipi/conf/load-balancer/baremetal-lab-ipi-conf-load-balancer-commands.sh#L16-L18

The current UPI one is at https://github.com/openshift/release/blob/54524b77de44726a55c48f371bcd3d916e957b5f/ci-operator/step-registry/baremetal/lab/pre/load-balancer/baremetal-lab-pre-load-balancer-commands.sh#L18-L40 and it should fit better this setup.


Therefore, you could create:
1. a new chain `baremetal-lab-ipi-conf-external-lb`, based on [baremeta-lab-ipi-conf](https://github.com/openshift/release/blob/519acedbb12e8f27cd576157e6a72860a35f5f0c/ci-operator/step-registry/baremetal/lab/ipi/conf/baremetal-lab-ipi-conf-chain.yaml), but using `baremetal-lab-pre-conf-load-balancer` instead of `baremetal-lab-ipi-conf-load-balancer`.
2. a new workflow `baremetal-lab-ipi-external-lb` based on [baremetal-lab-ipi](https://github.com/openshift/release/blob/50d6bac9d2f4f2f2178dff1e0568b4c8fcc4402f/ci-operator/step-registry/baremetal/lab/ipi/baremetal-lab-ipi-workflow.yaml) that consumes the new conf chain in (1).

The example test config should be:

```
tests:
- as: baremetal-ipi-ovn-ipv4-p1-f4
  cluster: build05
  cron: 0 0 * * *
  steps:
    cluster_profile: equinix-ocp-metal-qe
    dependencies:
      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:arm64-latest
    env:
      AUX_HOST: openshift-qe-bastion.arm.eng.rdu2.redhat.com
      DISCONNECTED: "false"
      IPI: "true"
      PROVISIONING_HOST: openshift-qe-bastion.arm.eng.rdu2.redhat.com
      PROVISIONING_NET_DEV: enP2p2s0f1
      architecture: arm64
      masters: "3"
      workers: "2"
    test:
    - ref: clusterbot-wait
    workflow: baremetal-lab-ipi-external-lb
```

This runs on arm64 for now, and should be ok for this feature. For running amd64 IPI we miss a node and we are currently blocked. Once it is available, you can decide whether you also want an amd64 conf, but this feature doesn't seem having any architecture-specific feature requiring it.